src/main/scala/uk/gov/nationalarchives/tdr/api/http/GraphQLServerHttp4s.scala

```diff
@@ -0,0 +1,156 @@
+package uk.gov.nationalarchives.tdr.api.http
+
+import cats.effect.IO
+import cats.implicits._
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.Logger
+import io.circe.Json
+import sangria.ast.Document
+import sangria.execution._
+import sangria.marshalling.ResultMarshaller
+import sangria.marshalling.circe._
+import slick.jdbc.JdbcBackend
+import uk.gov.nationalarchives.tdr.api.auth.{AuthorisationException, ValidationAuthoriser}
+import uk.gov.nationalarchives.tdr.api.consignmentstatevalidation.{ConsignmentStateException, ConsignmentStateValidator}
+import uk.gov.nationalarchives.tdr.api.db.repository._
+import uk.gov.nationalarchives.tdr.api.graphql.DataExceptions.InputDataException
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, DeferredResolver, ErrorCodes, GraphQlTypes}
+import uk.gov.nationalarchives.tdr.api.service._
+import uk.gov.nationalarchives.tdr.keycloak.Token
+
+import scala.concurrent.ExecutionContext
+
+class GraphQLServerHttp4s() {
```

Comment:
There is a lot of common code between this new class and the existing `GraphQlServer.class`.

To avoid this duplication think its worth defining a new trait that contains the common code, and have the two classes inherit from the trait. 

Something like this:

```
trait Server { 
  def handleException(...): HandledException  = { ... }
  ... etc ...
}
```